src/probnum/statespace/integrator.py

```diff
@@ -319,10 +322,19 @@ def discretise(self, dt):
             @ self.precon(dt).T
         )
         zero_shift = np.zeros(len(state_trans_mat))
+
+        # The Cholesky factor of the process noise covariance matrix of the IBM
+        # always exists, even for non-square root implementations.
+        proc_noise_cov_cholesky = (
+            self.precon(dt)
+            @ self.equivalent_discretisation_preconditioned.proc_noise_cov_cholesky
+        )
```

Comment:
Do we get any benefit from computing this Cholesky factor when not using the `sqrt` implementation?
Otherwise, we might make this conditional.

Reply:
Hmmm my thinking here was that this cholesky factor is created for the equivalent discretisation (preconditioned), so it was super weird not to translate it here, too. What do you think?

Reply:
Could we make the computation of the Cholesky factor - both in `equivalent_discretisation_preconditioned` and in `discretise` - conditional on `self.forward`/`backward_implementation`?

Reply:
yes, we could. Thought it only happens once, so the impact on runtime should be almost neglectible. Do you think that it is worth it nevertheless?